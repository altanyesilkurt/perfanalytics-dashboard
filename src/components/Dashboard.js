import React from 'react'
import LineChart from 'react-svg-line-chart'

const Dashboard = ({dataToDraw}) =>{
    const coordinateData= dataToDraw && dataToDraw.map((el,i) => ({
        x:i,y:el
    }))
    return(
        <div>
        <p>Time</p>
        <LineChart
            data={coordinateData || []}
        />
        <p>Count</p>
        </div>
    )
}

export default Dashboard
