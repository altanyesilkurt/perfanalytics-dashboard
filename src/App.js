import React, { useEffect, useState } from "react";
import Dashboard from "./components/Dashboard";
import "./App.scss";

//  const URL = "https://performance-analytic.herokuapp.com/"
 const URL = "http://localhost:3000/"

function App() {
  const [graphData, setGraphData] = useState();
  useEffect(() => {
    fetch(URL)
      .then((res) => res.json())
      .then((result) => {
        setGraphData(result);
      });
  }, []);
  return (
    <>
      <div className="dashboard-wrapper">
        <div>
          <h3>TTFB</h3>
          <Dashboard dataToDraw={graphData && graphData.ttfbList} />
        </div>
        <div>
          <h3>FCP</h3>
          <Dashboard dataToDraw={graphData && graphData.fcpList} />
        </div>
        <div>
          <h3>DOM LOAD</h3>
          <Dashboard dataToDraw={graphData && graphData.domLoadList} />
        </div>
        <div>
          <h3>WINDOW LOAD</h3>
          <Dashboard dataToDraw={graphData && graphData.windowLoadList} />
        </div>
      </div>
    </>
  );
}

export default App;