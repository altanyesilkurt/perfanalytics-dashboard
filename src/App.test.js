import { shallow } from "enzyme";
import Dashboard from './components/Dashboard';
import App from "./App";

it("renders without crashing App", () => {
  shallow(<App />);
});

describe('App', () => {
  it('should render App', () => {
    const component = shallow(<App />);

    expect(component).toMatchSnapshot();
  });
});

describe('Dashboard', () => {
  it('should render Dashboard', () => {
    const component = shallow(<Dashboard />);

    expect(component).toMatchSnapshot();
  });
});