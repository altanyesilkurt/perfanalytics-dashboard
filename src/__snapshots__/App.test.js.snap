// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App 1`] = `ShallowWrapper {}`;

exports[`Dashboard should render Dashboard 1`] = `ShallowWrapper {}`;

exports[`DrawGraphics should render DrawGraphics 1`] = `ShallowWrapper {}`;
